scripts/imgtool/image.py

```diff
@@ -246,14 +258,22 @@ def ecies_p256_hkdf(self, enckey, plainkey):
     def create(self, key, enckey, dependencies=None):
         self.enckey = enckey
 
-        if dependencies is None:
-            dependencies_num = 0
-            protected_tlv_size = 0
-        else:
-            # Size of a Dependency TLV = Header ('BBH') + Payload('IBBHI')
-            # = 16 Bytes
+        protected_tlv_size = 0
+
+        if self.security_counter is not None:
+            # TLV info header + security counter TLV
+            # Size of the security counter TLV: header ('HH') + payload ('I')
+            #                                   = 4 + 4 = 8 Bytes
+            protected_tlv_size = TLV_INFO_SIZE + TLV_SIZE + 4
+
+        if dependencies is not None:
+            if protected_tlv_size == 0:
+                # TLV info header
+                protected_tlv_size = TLV_INFO_SIZE
```

Comment:
I believe you could remove this `if` block from here and add one that increments after all protected TLVs are handled, so in the future new code that is added does not need to keep copy/pasting this.

Reply:
Yes indeed. I've already modified it locally. I have another patch that adds a new TLV to the protected area and I encountered with this problem.:)